components/connector/internal/revocation/loader.go

```diff
@@ -0,0 +1,109 @@
+package revocation
+
+import (
+	"context"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+type Loader interface {
+	Run(ctx context.Context)
+}
+
+type revocationListLoader struct {
+	revocationListCache Cache
+	manager             Manager
+	configMapName       string
+	reconnectInternal   time.Duration
+}
+
+func NewRevocationListLoader(revocationListCache Cache,
+	manager Manager,
+	configMapName string,
+	reconnectInternal time.Duration,
+) Loader {
+	return &revocationListLoader{
+		revocationListCache: revocationListCache,
+		manager:             manager,
+		configMapName:       configMapName,
+		reconnectInternal:   reconnectInternal,
+	}
+}
+
+func (rl *revocationListLoader) Run(ctx context.Context) {
+	rl.startKubeWatch(ctx)
+}
+
+func (rl *revocationListLoader) startKubeWatch(ctx context.Context) {
+	for {
+		select {
+		case <-ctx.Done():
+			log.Println("context cancelled, stop revocation config map watcher")
+			return
+		default:
+		}
+		watcher, err := connectWatch(rl.manager, rl.configMapName)
+		if err != nil {
+			log.Printf("Could not initialize watcher: %s. Sleep for %s and try again...", err, rl.reconnectInternal.String())
+			time.Sleep(rl.reconnectInternal)
+			continue
+		}
+		log.Println("Waiting for revocation list configmap events...")
+		for {
+			if !rl.processEvents(ctx, watcher.ResultChan()) {
+				// Cleanup any allocated resources
+				log.Printf("Stopping revocation configmap watcher")
+				watcher.Stop()
+				time.Sleep(rl.reconnectInternal)
+				break
+			}
+		}
+	}
+}
+
+func (rl *revocationListLoader) processEvents(ctx context.Context, events <-chan watch.Event) bool {
+	for {
+		select {
+		case <-ctx.Done():
+			return false
+		case ev, ok := <-events:
+			if !ok {
+				return false
+			}
+			switch ev.Type {
+			case watch.Added:
+				fallthrough
+			case watch.Modified:
+				log.Println("revocation list updated")
+				config, ok := ev.Object.(*v1.ConfigMap)
+				if !ok {
+					log.Println("Unexpected error: object is not configmap. Try again")
```

Comment:
Isn't this a configuration issue which should not be silently retrained since it will never succeed. IS there any case that this will be the case if everything around watch is properly configured?

Reply:
I don't expect it to be any different, but wanted to handle the possibility. We have the option to report something or just print an error message. There is also the possibility to kill the process of course and k8s will start a new one, which will have the same configuration issue

Reply:
Since it is part of the connector itself I agree that it is enough to log it.